R4R: staking spec upgrades / some renaming (#3357)
cosmos/cosmos-sdk: docs/spec/staking/state.md

Validators objects should be primarily stored and accessed by the
`OperatorAddr`, an SDK validator address for the operator of the validator. Two

> additional indexes are maintained per validator object in order to fulfill

Reviewer: ```suggestion
additional indices are maintained per validator object in order to fulfill
```

required lookups for slashing and validator-set updates. A third special index
(`LastValidatorPower`) is also maintained which however remains constant

> throughout each block, unlike the first two indexes which mirror the validator

Reviewer: ```suggestion
throughout each block, unlike the first two indices which mirror the validator
```

`"2006-01-02T15:04:05.000000000"`. Noteably This format: 

 - right pads all zeros
 - drops the time zone info (uses UTC) 

In all cases, the stored timestamp represents the maturation time of the queue
element. 

### UnbondingDelegationQueue

For the purpose of tracking progress of unbonding delegations the unbonding
delegations queue is kept. 

- UnbondingDelegation: `0x41 | format(time) -> []DVPair`

```
type DVPair struct {
	DelegatorAddr sdk.AccAddress
	ValidatorAddr sdk.ValAddress
}
```

### RedelegationQueue

For the purpose of tracking progress of redelegations the redelegation queue is
kept. 

- UnbondingDelegation: `0x42 | format(time) -> []DVVTriplet`

```
type DVVTriplet struct {
	DelegatorAddr    sdk.AccAddress
	ValidatorSrcAddr sdk.ValAddress
	ValidatorDstAddr sdk.ValAddress
}
```

### ValidatorQueue

For the purpose of tracking progress of unbonding validators the validator
queue is kept. 

- ValidatorQueueTime: `0x43 | format(time) -> []sdk.ValAddress`

The stored object as each key is an array of validator operator addresses from
which the validator object can be accessed.  Typically it is expected that only

> a single validator record will operate a given timestamp however it is possible

Reviewer: ```suggestion
a single validator record will be associated with a given timestamp however it is possible
```
